### ISSUE TITLE
Inconsistency with secondary keyboard actions.

### ISSUE THREAD

**Issue by oddlama** [rkkr/simple-keyboard#157]
This keyboard is really great, but I everytime when I am typing a longer text, I notice an inconsistency in the way secondary key actions work. Note that I am using the german key layout.

Every key shows its main secondary purpose in the top right corner. Pressing "q" longer will bring up the popup for alternative meanings, with "1" preselected. This seems consistent and is what I expect, because the label on the key reflects that properly. Same for "n" which brings up "!" by default but lets me select "ń" and "ñ" as tertiary actions (press and swipe).

But now do the same with the "o" "a" "s" or "u" key. Lets look at the "u" key. While the key reflect that "7" should be the default value, "ü" is selected when I long press, and "7" is only a tertiary action. Therefore, I always mistype when using these keys.

I'd be grear if the default was to honor the real secondary action with priority. I use numer much more frequently than umlauts (äöü). Maybe make a setting out of it, but I think the proper default should be to make umlauts tertiary.

Thanks for the good app!

**Comment by rkkr** [comment]
This was requested by @runiq and implemented @EmmanuelMess in https://github.com/rkkr/simple-keyboard/issues/91 . So you'll have to fight this out with them.

As I know the current functionality is also consistent with GBoard at least for other languages. Also the character on the top-right corner are not shown when you select "hide special characters". So the meaning of it is "this special char can be found here", not "default action for long-press".

**Comment by oddlama** [comment]
**TL;DR:** Can you make it an option?

Hi @rkkr, thanks for the reference to the other issue.

I just want to point out that even if "this special char can be found here" is the actual meaning, it is usually interpreted as "long-press action", because this is what it does for every key except 4 of them. So when I start using the keyboard I learn to associate "label shows long-press action". It's only until you hit one of the special ones when you notice that there are more options.

I am originally coming from "Hackers Keyboard", where the keys behave in the way that I consider "correct", although there probably is no real "correct" thing here.
I have also checked SwiftKey to see how a popular keyboard implements it, and they also do it this way. And I checked GBoard, and as you said it's like Simple Keyboard. But this is first time I've seen this behavior. (Which I personally find awkward to use)

Consider you want to quickly write a number like "19". I'd quickly long-press <kbd>q</kbd> to get the `1`, and naturally expect the <kbd>o</kbd> to give me a `9`, because the key label says so. But i get `1ö` which is annoying. One could argue "If you want to write numbers use the symbol key, type numbers, go back", but for short things like "I'm home at 9." it's definitely quicker to long-press. Also <kbd>a</kbd> results in `ä` instead of <kbd>@</kbd>.

I wouldn't consider to disable the key labels, as they are actually helpful. At least I cannot remember on which letter I would find any symbol like <kbd>"</kbd> <kbd>'</kbd> or <kbd>-</kbd>.

In the end, different people will have different preferences, as @ruinq and @EmmanuelMess for example. Not that this is a bad thing, apparently their use case is just different from mine. In my opinion the best compromise would be to make it an option. I really hope this is possible, but it's your decision after all, and otherwise your app is great.

**Comment by EmmanuelMess** [comment]
@oddlama I want to make it clear that I was just fixing a bug, and do not use this keyboard, nor speak German, therefore have nothing to add.

**Comment by oddlama** [comment]
@EmmanuelMess Yes, I was just reading your pull request and editing my answer to reflect that. You only fixed an inconsistency. Sorry for the inconvenience.

**Comment by rkkr** [comment]
As special char symbols in top-right corner will not be changed, closing this. If someone doesn't want German chars to be first in the popups, that can be a separate issue.
